docs/proposal/AI-gateway.md

```diff
@@ -0,0 +1,156 @@
+
+In this way, our AI plugin can be built as an external independent service and can be deployed on demand when AI traffic needs to be processed. At the same time, we can use any dev language, not necessarily C++. Of course, we use Golang. Also it is isolated from the complexity of Envoy to greatest extent, and only needs to process traffic requests/responses from gRPC connections. This fully ensures that development is simple, flexible and friendly.
+
+#### Integration with Existing Gateways
+
+The integration of various existing gateways and AI plugins is also very convenient. For example, Envoy Gateway provides a dedicated CRD for External Processing. As follows:
+
+```yaml
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: EnvoyExtensionPolicy
+metadata:
+  name: ext-proc-example
+  namespace: ollama
+spec:
+  targetRefs:
+    - group: gateway.networking.k8s.io
+      kind: HTTPRoute
+      name: ollama
+  extProc:
+  - backendRefs:
+    - name: aiengine
+      port: 9002
+    processingMode:
+      request:
+        body: Buffered
+      response:
+        body: Streamed
+
+```
+
+Istio Ingress Gateway can also use EnvoyFilter to add External Processing Filters on demand. Also, if necessary, as shown in the above arch picture, Envoy can be directly connected to AI plugin (it may no longer be appropriate to call it AI plugin), and then the AI plugin connects to the control plane of Gateway. The AI plugin modifies the XDS obtained from the control plane, for example, currently adding External Processing Filters, and finally sends the modified XDS to Envoy. This further improves the Gateway-agnosticity of the AI plugin and no additional configuration is required for different Gateway implementations.
+
+
+#### AI Plugin CRD
+
+Finally, we need to rely on CRD to describe the config related to AI traffic management. The config consists of the following two parts:
+
+1. Associated Service: The AI model is ultimately deployed in a service. We hope that only the service configured with the corresponding AI policy will forward traffic to the AI plugin and all AI services in the cluster can share the same AI plugin deployment without having to deploy one for each Service. In addition, if a service configured with AI policy, traffic can be directly forwarded to the AI plugin. Whether it's through automatically generating resource objects such as `EnvoyExtensionPolicy` or modifying XDS. It's believed that it will bring a better user experience.
+
+2. Specific AI traffic management policy: such as rate limit, cache, fallback, etc.
+
+The specific examples are as follows:
+
+```yaml

```

Comment:
Can you elaborate on this crd field?
Like https://github.com/martin-helmich/kubernetes-crd-example/blob/master/api/types/v1alpha1/project.go